### PR TITLE
Extend components

### PR DESCRIPTION
**Pull request by bleto**

### CODE REVIEW

File: src/templates/pluginVuex.ejs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+import deepmerge from 'deepmerge';
+import extendsModules from '~/.nuxt/extends.modules';
+import storeModules from '~/.nuxt/store.modules';
+
+export default async ({ app }) => {
+    const storesForExtend = {};
```
Comment (derpdead) [inline src/templates/pluginVuex.ejs]: storesToExtend - would be better naming option

File: src/templates/pluginVuex.ejs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    const asyncForEach = async (array, callback) => {
+        for (let index = 0; index < array.length; index += 1) {
+            await callback(array[index], index, array);
+        }
+    };
+
+    await asyncForEach(Object.values(extendsModules), async (extendContent) => {
```
Comment (derpdead) [inline src/templates/pluginVuex.ejs:24]: Could you explain me what is the idea behind this? Even if Vuex store registering is a Promise - tho it's not. Using Promisses will not speed it up. For each step you are wrapping getting value at indeax as a callback and awaiting for it so it's like do step 1, wait for result and do step 2 and wait for result. Other thing would be to do Promise.all

File: src/templates/pluginVuex.ejs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        if (!storeModules[storeName].mutations) {
+            storeModules[storeName].mutations = {};
+        }
+        if (!storeModules[storeName].actions) {
+            storeModules[storeName].actions = {};
+        }
+
+        if (storesForExtend[storeName]) {
+            if (storesForExtend[storeName].state) {
```
Comment (derpdead) [inline src/templates/pluginVuex.ejs]: You may Iterate via Object.keys(storesForExtend[storeName]) without checking for each individual

**Review by derpdead — CHANGES_REQUESTED** [review]

File: src/templates/pluginVuex.ejs (hunk 2 above)
Comment (bleto) [reply src/templates/pluginVuex.ejs:24]: In this place, I have situation where `extendContent.extendStore[extendName]()` return Promise and I need waiting for this. So I used await loop, and close this mechanism in one place. Promise.all() need waiting for array and then I must looping for this information, so I choose await loop mechanism.

**Review by derpdead — APPROVED** [review]

**Pull request merged**